models/music_cont_out_proxy.cpp

```diff
@@ -0,0 +1,416 @@
+{
+}
+
+void
+nest::music_cont_out_proxy::init_buffers_()
+{
+  B_.data_.clear();
+}
+
+void
+nest::music_cont_out_proxy::finalize()
+{
+}
+
+nest::port
+nest::music_cont_out_proxy::send_test_event( Node& target,
+  rport receptor_type,
+  synindex,
+  bool )
+{
+
+  DataLoggingRequest e( P_.interval_, P_.record_from_ );
+  e.set_sender( *this );
+  port p = target.handles_test_event( e, receptor_type );
+  if ( p != invalid_port_ and not is_model_prototype() )
+  {
+    B_.has_targets_ = true;
+  }
+
+  return p;
+}
+
+void
+nest::music_cont_out_proxy::calibrate()
+{
+  // only publish the output port once,
+  if ( S_.published_ == false )
+  {
+    const Token synmodel =
+      kernel().model_manager.get_synapsedict()->lookup( "static_synapse" );
+    assert(
+      synmodel.empty() == false && "synapse 'static_synapse' not available" );
+
+    const index synmodel_id = static_cast< index >( synmodel );
+    std::vector< long >::const_iterator t;
+
+    for ( t = P_.target_gids_.begin(); t != P_.target_gids_.end(); ++t )
+    {
+      // check whether the target is on this process
+      if ( kernel().node_manager.is_local_gid( *t ) )
+      {
+        Node* const target_node = kernel().node_manager.get_node( *t );
+        kernel().connection_manager.connect(
+          get_gid(), target_node, target_node->get_thread(), synmodel_id );
+      }
+    }
+    std::vector< MUSIC::GlobalIndex > music_index_map;
+    for ( size_t i = 0; i < P_.target_gids_.size(); i++ )
+    {
+      if ( kernel().node_manager.is_local_gid( P_.target_gids_[ i ] ) )
+      {
+        for ( size_t j = 0; j < P_.record_from_.size(); ++j )
+        {
+          music_index_map.push_back( P_.record_from_.size() * i + j );
+        }
+      }
+    }
+    MUSIC::Setup* s = kernel().music_manager.get_music_setup();
+    if ( s == 0 )
+    {
+      throw MUSICSimulationHasRun( get_name() );
+    }
+
+    MUSIC::ContOutputPort* MP = s->publishContOutput( P_.port_name_ );
+
+    if ( MP->isConnected() == false )
+    {
+      throw MUSICPortUnconnected( get_name(), P_.port_name_ );
+    }
+
+    if ( MP->hasWidth() == false )
+    {
+      throw MUSICPortHasNoWidth( get_name(), P_.port_name_ );
+    }
+
+    S_.port_width_ = MP->width();
+    const size_t per_port_width = P_.record_from_.size();
+
+    // Allocate memory
+    B_.data_.resize( per_port_width * S_.port_width_ );
+
+    // Check if any port is out of bounds
+    if ( P_.target_gids_.size() > S_.port_width_ )
+    {
+      throw MUSICChannelUnknown(
+        get_name(), P_.port_name_, S_.port_width_ + 1 );
+    }
+
+    // The permutation index map, contains global_index[local_index]
+    MUSIC::PermutationIndex* music_perm_ind = new MUSIC::PermutationIndex(
+      &music_index_map.front(), music_index_map.size() );
+
+    MUSIC::ArrayData* dmap =
+      new MUSIC::ArrayData( static_cast< void* >( &( B_.data_.front() ) ),
+        MPI::DOUBLE,
+        music_perm_ind );
+
```

Comment:
I will fix this. But just for the protocol, you already approved it :) and I just merged it into my branch.